src/test/kotlin/com/forgerock/uk/openbanking/support/payment/PaymentApiClient.kt

```diff
@@ -85,6 +101,72 @@ class PaymentApiClient(val tpp: Tpp) {
             }
             return result.get()
         }
+
+        inline fun sendFileRequest(): Boolean {
+            if (jwsSignatureProducer != null && jsonBody != null) {
```

Comment:
It seems a duplicated method to me, the only difference is the content type, could be better create only one method and pass the content type as parameter like:
```kotlin
import org.apache.http.entity.ContentType
import com.github.kittinunf.fuel.core.Headers
sendFileRequest(contentType: ContentType) {  // ContentType.APPLICATION_JSON or ContentType.TEXT_XML
.....
request.header(Headers.CONTENT_TYPE, contentType)
}
````

Reply:
We have refactored according to the requested changes.